src/lib/time-formats.js

```diff
@@ -1,14 +1,50 @@
-/* exported formats */
-
-export const formats = {
-  default: {
-    short_time: '[@] ha',
-    short_date_time: 'ddd [@] ha',
-    long_date_time: 'ddd MMM D \ [@] ha'
-  },
-  'it': {
-    short_time: '[@] H',
-    short_date_time: 'ddd [@] H',
-    long_date_time: 'ddd D MMM \ [@] H'
+/* exported getLocalizedDateTime */
+
+const uiLocale = [browser.i18n.getUILanguage().replace('_', '-'), 'en-US'];
+
+const formats = {
+  'date_day': new Intl.DateTimeFormat(uiLocale, { weekday: 'short', month: 'short', day: 'numeric' }),
+  'date_year': new Intl.DateTimeFormat(uiLocale, { month: 'short', day: 'numeric', year: 'numeric' }),
+  'long_date_time': new Intl.DateTimeFormat(uiLocale, { weekday: 'short', month: 'short', day: 'numeric', hour: 'numeric', minute: 'numeric' }),
+  'short_date_time': new Intl.DateTimeFormat(uiLocale, { weekday: 'short', hour: 'numeric', minute: 'numeric' }),
+  'short_time': new Intl.DateTimeFormat(uiLocale, { hour: 'numeric', minute: 'numeric' }),
+
+  'confirmation_time': new Intl.DateTimeFormat(uiLocale, { hour: 'numeric', minute: 'numeric' }),
+  'confirmation_time_no_minutes': new Intl.DateTimeFormat(uiLocale, { hour: 'numeric', minute: 'numeric' }),
+  'confirmation_time_no_minutes-en': new Intl.DateTimeFormat(uiLocale, { hour: 'numeric' }),
+  'confirmation_date': new Intl.DateTimeFormat(uiLocale, { weekday: 'short', month: 'short', day: 'numeric' }),
+
+  'long_date_time-en': [
+    new Intl.DateTimeFormat(uiLocale, { weekday: 'short', month: 'short', day: 'numeric' }),
+    new Intl.DateTimeFormat(uiLocale, { hour: 'numeric' })
+  ],
+  'short_date_time-en': [
+    new Intl.DateTimeFormat(uiLocale, { weekday: 'short' }),
+    new Intl.DateTimeFormat(uiLocale, { hour: 'numeric' })
+  ],
+  'short_time-en': [
+    {format: () => ''},
+    new Intl.DateTimeFormat(uiLocale, { hour: 'numeric' })
+  ],
+};
+
+export const getLocalizedDateTime = function (time, format) {
+  let formattedDateTime;
+
+  let formatter = formats[format];
+  if (formatter) {
+    const realLocale = formatter.resolvedOptions().locale;
+    formattedDateTime = formatter.format(time.toDate());
```

Comment:
I wonder if this should be a return after the if instead, since you discard this value for en-* locales